### ISSUE TITLE
Murkens archive bug

### ISSUE THREAD

**Issue by NumberM** [keyteki/keyteki#216]
doesn't play random card from archive when the opponents draw pile is at 0

**Comment by Herzbube76** [comment]
Murkens has bugs which cant be reproduced. Sometimes works, sometimes not.

1. game: wanted to choose archive (2 cards) but was forced to play top of deck...
2. game: archive of opponent was empty and nothing was played from deck (was 2. turn so enough cars here). 2 turns later it did work correctly (no archive and it did play top of deck automatically)

**Comment by Herzbube76** [comment]
One more bug, which now happend twice to me with murkens:

if i play persistance hunting from opponent by murkens, the creatures of the chosen house arent exhausted, always need to be manualed.

**Comment by MythicMikeneto** [comment]
I tried to play a card from the opponents archive (he had 8 cards in there) and it did nothing.

It's possible I hit an Alpha card, but there is no way to tell.

**Comment by hydrophilic-attack** [comment]
I had a game today where my opponent played murkens and chose my archives and nothing happened

My guess is that nothing happened because there was a 50% chance for an alpha card to be chosen, and that is what happened. But if that is the case, the log should have explained it. The log said nothing after murkens was played

**Comment by CrookedWookie** [comment]
I played Murkens to play a random card from my opponent's archive; it played Burn the Stockpile when he DID actually have 7 aember in his pool.  Jackpot.  Only it didn't play the card; no aember was lost, the card didn't leave his archive and enter his discard pile.  We had to fix the whole thing in manual.

**Comment by hydrophilic-attack** [comment]
Played a game just now where I played Murkens and chose their archives, where they had 4 cards. According to the log, I played their Hexpion, but nothing happened. It stayed in their archive.

We tried to fix it in manual mode as well but that didn't work either. My opponent moved Hexpion manually from their archives to their play area, but nothing happened when they tried to give control to me. After trying, the Hexpion was still shown in their play area, but wasn't actually there.

The game log for the game is attached
[murkens_hexpion.txt](https://github.com/keyteki/keyteki/files/3483852/murkens_hexpion.txt)

**Comment by romanogit** [comment]
Today I played and the opponent had 4 cards in the archive and a few on his deck.
Among the 4 cards, 3 were mine.
When I played Murkens, the engine forced me to get a card from the archive.

**Comment by romanogit** [comment]
A new bug with murchens. Opponent had 2 cards in his archive, an hexpion and one more. I play Murkens and selected archive. Log message says I would play Hexpion, but Titan was played instead:

```
romanomagacho22 plays Murkens
romanomagacho22 uses Murkens to play Hexpion
romanomagacho22 plays Titan Mechanic
```

**Comment by romanogit** [comment]
I think one of the issues with Murkens is the shuffleness. I think it is calling the targets calculation function multiple times. Maybe one to evaluate the need to prompt and, if not needed, it applies again when playing the card.

```
'Random card from archives': [
                        ability.actions.playCard(context => ({ target: _.shuffle(context.player.opponent.archives)[0] }))
                    ]
```

**Comment by cryogen** [comment]
Lots of mentions in this issue about alpha cards being confusing, that’s now fixed. It should reveal and then not play it

**Comment by xjoaniex** [comment]
My opponent's deck had 0 cards.  
Their archive had 6 cards.I played Murkens and was not able to get one of the cards from the archive.
There was no choice presented for where to get the card from.
Basically I played Murkens and nothing happened.

Is there a bug that occurs when either the deck or archive is empty?

![image](https://user-images.githubusercontent.com/3081800/85554178-a8631400-b5f2-11ea-9a3c-23328bfd03a9.png)
